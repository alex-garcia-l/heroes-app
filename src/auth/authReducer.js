import { authType } from '../types/auth';

export const authReducer = (state = {}, action) => {
  
  switch (action.type) {
    case authType.login:
      return {
        ...action.payload,
        logged: true
      }
    case authType.logout:
      return {
        logged: false
      }
  
    default:
      return state;
  }

}