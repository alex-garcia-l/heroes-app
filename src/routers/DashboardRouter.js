import React from 'react';
import { Route, Routes } from 'react-router-dom';

import { DcScreen } from '../components/dc/DcScreen';
import { HeroScreen } from '../components/heros/HeroScreen';
import { HomeScreen } from '../components/HomeScreen';
import { MarvelScreen } from '../components/marvel/MarvelScreen';
import { SearchScreen } from '../components/search/SearchScreen';

import { Navbar } from '../components/ui/Navbar';

export const DashboardRouter = () => {
  return (
    <>
      <Navbar />
      <div className="container mt-4">
        <Routes>
          <Route exact index path="/" element={<HomeScreen />} />
          <Route exact path="marvel" element={<MarvelScreen />} />
          <Route exact path="search" element={<SearchScreen />} />
          <Route exact path="hero/:id" element={<HeroScreen />} />
          <Route exact path="dc" element={<DcScreen />} />
        </Routes>
      </div>
    </>
  );
};
