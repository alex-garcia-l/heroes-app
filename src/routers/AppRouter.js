import React from 'react';
import { BrowserRouter, Route, Routes } from 'react-router-dom';

import { PrivateRoute } from './PrivateRoute';
import { PublicRoute } from './PublicRoute';

import { LoginScreen } from '../components/login/LoginScreen';
import { DashboardRouter } from './DashboardRouter';

export const AppRouter = () => {
  return (
    <BrowserRouter>
      <Routes>

        <Route
          exact
          path="login"
          element={
            <PublicRoute>
              <LoginScreen />
            </PublicRoute>
          }
        />

        <Route path="/*" 
          element={
            <PrivateRoute>
              <DashboardRouter />
            </PrivateRoute>
          }
        />
      </Routes>
    </BrowserRouter>
  );
};
