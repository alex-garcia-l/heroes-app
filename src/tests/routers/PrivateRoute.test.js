import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import { AuthContext } from '../../auth/authContext';
import { PrivateRoute } from '../../routers/PrivateRoute';

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  Navigate: () => <h2>Saliendo del componente</h2>
}));

describe('Prueba para el componente <PrivateRoute />', () => {

  Storage.prototype.setItem = jest.fn();

  test('debe mostrar el componente si está autenticado y guardar en el localstorage', () => {

    const contextValue = {
      user: {
        logged: true
      }
    }

    const text = 'Private Component';
    const path = '/';

    const wrapper = mount(
      <AuthContext.Provider value={contextValue}>
        <MemoryRouter initialEntries={[path]}>
          <PrivateRoute>
            <h1>{text}</h1>
          </PrivateRoute>
        </MemoryRouter>
      </AuthContext.Provider>
    );

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('h1').text().trim()).toBe('Private Component');
    expect(localStorage.setItem).toHaveBeenCalledWith('last_path', path);
  });

  test('debe bloquear al componente si no está autenticado', () => {
    const contextValue = {
      user: {
        logged: false
      }
    }

    const text = 'Private Component';
    const path = '/';

    const wrapper = mount(
      <AuthContext.Provider value={contextValue}>
        <MemoryRouter initialEntries={[path]}>
          <PrivateRoute>
            <h1>{text}</h1>
          </PrivateRoute>
        </MemoryRouter>
      </AuthContext.Provider>
    );

    expect(wrapper.find('h2').text().trim()).toBe('Saliendo del componente');
  });

});
