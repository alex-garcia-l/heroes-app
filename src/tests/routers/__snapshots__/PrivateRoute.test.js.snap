// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba para el componente <PrivateRoute /> debe mostrar el componente si est√° autenticado y guardar en el localstorage 1`] = `
<h1>
  Private Component
</h1>
`;
